File: src/main/java/com/stratio/connector/hdfs/connection/HDFSConnectionHandler.java
package com.stratio.connector.hdfs.connection;

import com.stratio.connector.commons.connection.Connection;
import com.stratio.connector.commons.connection.ConnectionHandler;

import com.stratio.crossdata.common.connector.ConnectorClusterConfig;
import com.stratio.crossdata.common.connector.IConfiguration;
import com.stratio.crossdata.common.exceptions.ConnectionException;
import com.stratio.crossdata.common.security.ICredentials;

/**
 * Class implements native HDFS connection.
 *
 */
public class HDFSConnectionHandler extends ConnectionHandler {


    public HDFSConnectionHandler(IConfiguration configuration) {
        super(configuration);
    }

    /**
     * Use config & Credentials to create HDFS native connection.
     *
     * @param iCredentials
     *            .
     * @param connectorClusterConfig
     *            .
     *
     * @return DeepConnection.
     **/
    @Override
    protected Connection createNativeConnection(ICredentials iCredentials,
            ConnectorClusterConfig connectorClusterConfig) throws ConnectionException {


            return new HDFSConnection(iCredentials,connectorClusterConfig);

    }
}

File: src/main/scala/com.stratio.connector.hdfs.scala/connection/HDFSConnectionHandler.scala
package com.stratio.connector.hdfs.scala.connection

import com.stratio.connector.commons.connection.{Connection, ConnectionHandler}
import com.stratio.crossdata.common.connector.{ConnectorClusterConfig, IConfiguration}
import com.stratio.crossdata.common.security.ICredentials

class HDFSConnectionHandler(iConfiguration: IConfiguration)
  extends ConnectionHandler(iConfiguration){
  override def createNativeConnection(
    credentials: ICredentials,
    config: ConnectorClusterConfig): Connection[_] =
    new HDFSConnection(credentials,config)
}
